core/tauri/src/updater/core.rs

```diff
@@ -640,14 +635,28 @@ impl<R: Runtime> Update<R> {
 // We should have an AppImage already installed to be able to copy and install
 // the extract_path is the current AppImage path
 // tmp_dir is where our new AppImage is found
-#[cfg(feature = "updater")]
 #[cfg(target_os = "linux")]
 fn copy_files_and_run<R: Read + Seek>(archive_buffer: R, extract_path: &Path) -> Result {
-  use std::os::unix::fs::PermissionsExt;
-  let tmp_dir = tempfile::Builder::new()
+  use std::os::unix::fs::{MetadataExt, PermissionsExt};
+
+  let extract_path_metadata = extract_path.metadata()?;
+
+  let mut tmp_dir = tempfile::Builder::new()
     .prefix("tauri_current_app")
     .tempdir()?;
-  let mut perms = std::fs::metadata(tmp_dir.path())?.permissions();
+  let mut tmp_dir_metadata = tmp_dir.path().metadata()?;
+
+  // fallback to cache_dir or CWD if the tmp directory is not in the same mount point
+  if extract_path_metadata.dev() != tmp_dir_metadata.dev() {
+    let fallback_dir =
+      dirs_next::cache_dir().unwrap_or_else(|| extract_path.parent().unwrap().to_path_buf());
+    tmp_dir = tempfile::Builder::new()
+      .prefix("tauri_current_app")
+      .tempdir_in(fallback_dir)?;
+    tmp_dir_metadata = tmp_dir.path().metadata()?
```

Comment:
It doesn't look like we're checking the metadata again to be sure those are on the same mountpoint?
Especially if `~/.cache` is found and preferred over the current `.appimage` path, this seems quite possible.
I could be running the `appimage` from a network share with `~/.cache` on a local btrfs volume.

Reply:
You're right. I'll see how I can change that.

Reply:
done @Beanow code might look weird with the vector of closures, but i think it's the way to remove duplicated code lol

Reply:
Yeah that looks like a good approach to me :ok_hand: